airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -499,4 +515,24 @@ protected static String toSslJdbcParamInternal(final SslMode sslMode) {
     return result;
   }
 
+  @Override
+  protected boolean verifyCursorColumnValues(final JdbcDatabase database, final String schema, final String tableName, final String columnName) throws SQLException {
+    final String query;
+    final String resultColName = "nullValue";
+    // Query: Only if cursor column allows null values, query whether it contains one
+    if (StringUtils.isNotBlank(schema)) {
+      query = String.format(NULL_CURSOR_VALUE_WITH_SCHEMA,
+          schema, tableName, columnName, schema, tableName, columnName, resultColName);
+    } else {
+      query = String.format(NULL_CURSOR_VALUE_NO_SCHEMA,
+          tableName, columnName, tableName, columnName, resultColName);
+    }
+    LOGGER.debug("null value query: {}", query);
+    final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(conn -> conn.createStatement().executeQuery(query),
```

Comment:
Just curious - why not a prepared statement here?

Reply:
I _think_ it is safer with a prepared statement with regards to escaping special characters?
But I may remember it incorrectly from the past.
Anything I can improve here?

Reply:
They're normally preferred since:

- They are pre-compiled (so generally faster)
- Prevent SQL injection - so protect against bad table name/column name/etc in this case

Reply:
Prepared statement is a pain here as the query also parameterizes schema/table/column names which are not compatible - PS adds a single quote which causes a syntax error.

Reply:
Ac